src/vs/workbench/contrib/chat/common/chatModel.ts

```diff
@@ -180,9 +180,19 @@ export interface IChatTextEditGroup {
 	done: boolean | undefined;
 }
 
+export function isCellTextEditOperation(value: unknown): value is ICellTextEditOperation {
+	const candidate = value as ICellTextEditOperation;
+	return !!candidate && !!candidate.edit && !!candidate.uri && URI.isUri(candidate.uri);
+}
+
+export interface ICellTextEditOperation {
+	edit: TextEdit;
+	uri: URI;
+}
+
 export interface IChatNotebookEditGroup {
 	uri: URI;
-	edits: ICellEditOperation[][];
+	edits: (ICellTextEditOperation | ICellEditOperation)[];
```

Comment:
Text edits belonging to a notebook must belong to `notebookGroup` 
This way the order is preserved and its clear these text edits belong to notebooks.